internal/generator/generator.go

```diff
@@ -14,7 +14,8 @@ import (
 // Generate receives a string sourcePath and an io.Writer writer. It will
 // extract all the directives definitions from the .c and .cpp files in
 // sourcePath and its subdirectories, then output the corresponding directive
-// masks map named "directives" and matchFunc named "Match" via writer.
-func Generate(sourcePath string, writer io.Writer, filter map[string]struct{}, override map[string][]Mask) error {
-	return genFromSrcCode(sourcePath, "directives", "Match", writer, filter, override)
+// masks map and matchFunc via writer.
+func Generate(sourcePath string, writer io.Writer, filter map[string]struct{}, override map[string][]Mask,
+	directiveMapName string, matchFuncName string) error {
+	return genFromSrcCode(sourcePath, directiveMapName, matchFuncName, writer, filter, override)
```

Comment:
strong stylistic nit-- if you need to make a function header multiple lines the return value should always be it's own line, never shared with params. See how at the moment `directiveMapName` and `return` are at the same indent level? That makes it harder to quickly identify what is function body vs parameters (in this case `return` is a pretty big hint though).

If we rearrange slightly:
```suggestion
func Generate(
        sourcePath string, 
        writer io.Writer, 
        filter map[string]struct{}, 
        override map[string][]Mask, 
        directiveMapName string, 
        matchFuncName string,
) error {
	return genFromSrcCode(sourcePath, directiveMapName, matchFuncName, writer, filter, override)
```

Now the func header is _very_ clearly delineated.  Occasionally you can group multiple params onto one line-- often with many params there are subgroups that go together.  Conversely many params is a bit of a smell itself.

Reply:
Oh I see. Multiple lines seems better. 

> Conversely many params is a bit of a smell itself.

Yeah I'm quite unsure if I should pack them into a struct here... Is there a convention about the maximum number of params in our group?